### PR TITLE
moka-icon-theme: 5.3.6 -> 5.4.0, faba-icon-theme: 2016-09-13 -> 4.3

### PR DESCRIPTION
**Pull request by romildo**
###### Motivation for this change

- Update `moka-icon-theme` to version 5.4.0
- Update `faba-icon-theme` to version 4.3
- Build with meson
- Remove explicitly building of the icon cache, as it is built automatically now
- Restrict `moka-icon-theme` (and its dependents) to linux platforms, because the distributed archive contains file names that differ only in case, which is an issue for darwin

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mic92** [comment]
@GrahamcOfBorg build  moka-icon-theme

File: pkgs/data/icons/moka-icon-theme/default.nix [hunk 1]
```diff
@@ -1,24 +1,18 @@
...
   pname = "moka-icon-theme";
-  version = "5.3.6";
+  version = "5.4.0";
 
   src = fetchFromGitHub {
     owner = "snwh";
     repo = pname;
     rev = "v${version}";
-    sha256 = "17f8k8z8xvib4hkxq0cw9j7bhdpqpv5frrkyc4sbyildcbavzzbr";
+    sha256 = "015l02im4mha5z91dbchxf6xkp66d346bg3xskwg0rh3lglhjsrd";
```
Comment (Mic92) [inline pkgs/data/icons/moka-icon-theme/default.nix]: ofBorg gets `166wanxa4azqff6h0cw0n98c3gc4zlvjynmigz8v7cq5d4s7jg9h` as a checksum here.

Comment (romildo) [reply pkgs/data/icons/moka-icon-theme/default.nix]: It seems that the checksum differ on darwin and linux.

**Comment by romildo** [comment]
>Failure on x86_64-linux

It works on my NixOS system.

**Comment by romildo** [comment]
Hash mismatch on darwin looks like another instance of #39308, probably due to case conflicts.

Restricting to linux platforms to avoid this issue.

**Comment by romildo** [comment]
@grahamcofborg build faba-icon-theme moka-icon-theme arc-icon-theme faba-mono-icons

**Comment by romildo** [comment]
@GrahamcOfBorg build faba-icon-theme moka-icon-theme arc-icon-theme faba-mono-icons

**Comment by jtojnar** [comment]
You will most definitely need to add `patchShebangs meson/post_install.py` in `postPatch`. It is pretty common fix with meson projects.

How did you determine that sandbox is working? It would not work in sandbox without patching the shebang.

**Comment by romildo** [comment]
> How did you determine that sandbox is working?

I have used `nix-build --option sandbox true` in the command line without being the `root` user. I have just found that this does not work unless the user is a trusted user.

After appropriately setting the user (which is in the `wheel` group) as a trusted user by adding `nix.trustedUsers = [ "root" "@wheel" ]` to `configuration.nix` in NixOS, and patching, it works.

**Comment by romildo** [comment]
@GrahamcOfBorg build faba-icon-theme moka-icon-theme

**Comment by romildo** [comment]
@GrahamcOfBorg build faba-icon-theme moka-icon-theme arc-icon-theme faba-mono-icons

**Comment by romildo** [comment]
@Mic92 Maybe this PR can be merged now.

**Pull request merged**